Lib/compression/zstd/_zstdfile.py

```diff
@@ -0,0 +1,378 @@
+import builtins
+import io
+
+from os import PathLike
+
+from _zstd import ZstdCompressor, ZstdDecompressor, _ZSTD_DStreamSizes, ZstdError
+from compression._common import _streams
+
+__all__ = ("ZstdFile", "open")
+
+_ZSTD_DStreamOutSize = _ZSTD_DStreamSizes[1]
+
+_MODE_CLOSED = 0
+_MODE_READ = 1
+_MODE_WRITE = 2
+
+
+class ZstdFile(_streams.BaseStream):
+    """A file object providing transparent zstd (de)compression.
+
+    A ZstdFile can act as a wrapper for an existing file object, or refer
+    directly to a named file on disk.
+
+    Note that ZstdFile provides a *binary* file interface - data read is
+    returned as bytes, and data to be written should be an object that
+    supports the Buffer Protocol.
+    """
+
+    _READER_CLASS = _streams.DecompressReader
+
+    FLUSH_BLOCK = ZstdCompressor.FLUSH_BLOCK
+    FLUSH_FRAME = ZstdCompressor.FLUSH_FRAME
+
+    def __init__(
+        self,
+        filename,
```

Comment:
Given it can accept file pointers, `filename` isn't quite accurate. Also make it positional-only
```suggestion
        file, /,
```